### ISSUE TITLE
Implement lazy instantiation of self._ssm, an instance of a boto3 ssm client.

### ISSUE THREAD

**Issue by JoshuaSchlichting** [MarletteFunding/omelette#24]
We would like to be able to use/reference [`omelette.Settings`](https://github.com/MarletteFunding/omelette/blob/main/omelette/settings.py) within offline tests that have no intent on talking to AWS. Currently, the [`Settings`](https://github.com/MarletteFunding/omelette/blob/main/omelette/settings.py) class will initialize a `boto3` SSM client upon instantiation, even if it isn't needed (no SSM params in `settings.toml`). The initialization of a `boto3` client immediately sends the library looking for AWS related content in the environment it's executing in.

For example, [`Settings`](https://github.com/MarletteFunding/omelette/blob/main/omelette/settings.py) has a dependency on the existence of an environment variable named `AWS_DEFAULT_REGION` because of this call to `boto3` in [`Settings.__init__`](https://github.com/MarletteFunding/omelette/blob/bc203dda582ec9720f91f93284a56365e04ec97e/omelette/settings.py#L48). The lack of this environment variable alone will cause offline unit tests, which don't touch AWS, to fail. Even if you pass this environment variable into the testing environment, you'll still fail with:
```
.local/lib/python3.9/site-packages/omelette/settings.py:51: in __init__
    self._ssm = boto3.client("ssm") if _boto_available else None
.local/lib/python3.9/site-packages/boto3/__init__.py:93: in client
    return _get_default_session().client(*args, **kwargs)
.local/lib/python3.9/site-packages/boto3/session.py:270: in client
    return self._session.create_client(
.local/lib/python3.9/site-packages/botocore/session.py:870: in create_client
    client = client_creator.create_client(
.local/lib/python3.9/site-packages/botocore/client.py:91: in create_client
    client_args = self._get_client_args(
.local/lib/python3.9/site-packages/botocore/client.py:360: in _get_client_args
    return args_creator.get_client_args(
.local/lib/python3.9/site-packages/botocore/args.py:99: in get_client_args
    endpoint = endpoint_creator.create_endpoint(
.local/lib/python3.9/site-packages/botocore/endpoint.py:335: in create_endpoint
    raise ValueError("Invalid endpoint: %s" % endpoint_url)
E   ValueError: Invalid endpoint: https://ssm..amazonaws.com
```

This is all when running tests with `[local]` settings, without any references to SSM, the `Settings`.

# **tl;dr**
Let's initialize `self._ssm` to `None`, and then check state and initialize if needed when [`_set_from_ssm()`](https://github.com/MarletteFunding/omelette/blob/bc203dda582ec9720f91f93284a56365e04ec97e/omelette/settings.py#L167) is called, so that [`omelette.Settings`](https://github.com/MarletteFunding/omelette/blob/main/omelette/settings.py) can be used for reading toml files that are unrelated to AWS SSM.
